src/python/pants/option/config.py

```diff
@@ -290,11 +302,8 @@ def get_value(self, section: str, option: str) -> str | None:
         return remove_val
 
     @property
-    def defaults(self) -> dict[str, str]:
-        return {
-            option: self._stringify_val_without_interpolation(option_val)
-            for option, option_val in self.section_to_values[DEFAULT_SECTION].items()
-        }
+    def defaults(self) -> dict[str, Any]:
```

Comment:
After staring at this and related code for a while, I think I've convinced myself that not stringifying here is fine. It makes me a little nervous, but the tests are comprehensive, so I will trust to them.

Reply:
Yeah, its private-single-use only (except for the config_test):
```python
possible_interpolations = {**self.defaults, **section_values}
return new_style_format_str.format(**possible_interpolations)
```

Originally I was populating each `env_X` separately in the seed values and then doing a `s/_/./` dance in the interpolations. I mistakenly thought `.` access only worked in f-strings. But it turns out `.` and `[]` access is documented to (and actually does) work in `.format(**named_substiturions)`:
```
$ python2.7 -c "import sys; print('{x.version_info[0]}'.format(x=sys))"
2
$ python3.10 -c "import sys; print('{x.version_info[0]}'.format(x=sys))"
3
```
Ao I went with the `env` `SimpleNamespace` object.